src/docs/scripts/reference.py

```diff
@@ -50,7 +52,21 @@
 from django_components.component import Component
 from django_components.templatetags.component_tags import TagSpec
 from django_components.util.misc import get_import_path
-from docs.scripts.extensions import _format_source_code_html
+
+# NOTE: This file is an entrypoint for the `gen-files` plugin in `mkdocs.yml`.
+#       However, `gen-files` plugin runs this file as a script, NOT as a module.
+#       That means that:
+#         - By default we can't do relative imports (e.g. `.extensions`)
+#         - We can't import from packages found in the `src` directory (e.g. `docs.scripts.extensions`)
+#
+#       But we need to import from another module.
+#       Hence we add the `src` directory to `sys.path` ourselves, so we can import from the
+#       `docs.scripts.extensions` module.
+current_dir = os.path.dirname(os.path.abspath(__file__))
+src_dir = os.path.abspath(os.path.join(current_dir, "..", ".."))
+sys.path.insert(0, src_dir)
+
+from docs.scripts.extensions import _format_source_code_html  # noqa: E402
```

Comment:
As the docs say, I had to modify the `sys.path`, so that the script can import from other files in `src/docs/scripts`.

I came across this only now, as it turns out I had pip-installed django-components in my django-components project, which made the python import paths under `docs` valid.

But when I started with a clean install, this script could no longer import from `docs`. Hence the workaround.

Reply:
Was it this missing sys.path modification that made you move the docs under src? (which is unusual)-  If so you could make docs a root folder again like in most other projects to not break expectations.

Reply:
Not sure I understand the wording, but my original intention to move the docs under `src` was to avoid having to do hacks with `sys.path`. But turns out I had to do it anyway. So yeah, the `docs` dir could be moved back to where it was before.

@dalito Would you do write up the MR for moving the `docs` dir back up (after this MR is merged)? There's still two other bugs after introducing v0.110 that I want to focus on first.

Reply:
Ok. I'll look into moving the docs to root after this is merged.